File: aat-android/src/main/java/ch/bailu/aat/description/SpeedDescription.java
package ch.bailu.aat.description;

import ch.bailu.aat_lib.description.FF;
import ch.bailu.aat_lib.description.FloatDescription;
import ch.bailu.aat_lib.preferences.StorageInterface;
import ch.bailu.aat_lib.preferences.general.SolidUnit;


public abstract class SpeedDescription extends FloatDescription {

    private final SolidUnit sunit;

    public SpeedDescription(StorageInterface storage) {
        sunit = new SolidUnit(storage);
    }

    @Override
    public String getUnit() {
        return sunit.getSpeedUnit();
    }


    @Override
    public String getValue() {
        float speed = getCache();
        float speedFactor = sunit.getSpeedFactor();
        speed = speed * speedFactor;
        return FF.f().N1.format(speed);
    }


    public String getSpeedDescription(float value) {
        setCache(value);
        return getValue() + getUnit();
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/SpeedDescription.kt
package ch.bailu.aat_lib.description

import java.text.NumberFormat
import ch.bailu.aat_lib.preferences.StorageInterface
import ch.bailu.aat_lib.preferences.general.SolidUnit

abstract class SpeedDescription(storage: StorageInterface, private val format: NumberFormat=FormatDisplay.f().decimal1) : FloatDescription() {
    private val sunit: SolidUnit = SolidUnit(storage)

    override fun getUnit(): String {
        return sunit.speedUnit
    }

    override fun getValue(): String {
        val speedFactor = sunit.speedFactor
        val speed = cache* speedFactor
        return format.format(speed.toDouble())
    }

    fun getSpeedDescription(value: Float): String {
        setCache(value)
        return getValue() + " " + getUnit()
    }
}
